cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/AppLifecycleHttpHandler.java

```diff
@@ -443,14 +503,33 @@ private void stopProgramIfRunning(Id.Program programId) throws InterruptedExcept
     return namespace;
   }
 
-  private Id.Application validateApplicationId(String namespaceId, String appId)
+  private ApplicationId validateApplicationId(String namespaceId, String appId)
     throws BadRequestException, NamespaceNotFoundException {
+    NamespaceId namespace = validateNamespace(namespaceId);
+    if (appId == null) {
+      throw new BadRequestException("Path parameter app-id cannot be empty");
+    }
+    if (EntityId.isValidId(appId)) {
+        return namespace.app(appId);
+    }
+    throw new BadRequestException(String.format("Invalid app name '%s'", appId));
+  }
 
-    Id.Namespace namespace = validateNamespace(namespaceId);
-    try {
-      return Id.Application.from(namespace, appId);
-    } catch (IllegalArgumentException e) {
-      throw new BadRequestException(String.format("Invalid app name '%s': %s", appId, e.getMessage()));
+  private ApplicationId validateApplicationVersionId(String namespaceId, String appId, String versionId)
+    throws BadRequestException, NamespaceNotFoundException {
+    NamespaceId namespace = validateNamespace(namespaceId);
+    if (appId == null) {
```

Comment:
That method directly returns an ApplicationId with a default version.  I've also been thinking merging them. How about calling `validateApplicationVersionId (namespaceId, appId, DEFAULT_VERSION)` from `validateApplicationId`?